projects/packages/block-delimiter/README.md

```diff
@@ -6,10 +6,233 @@ Efficiently work with block structure
 
 To use this package in your WordPress plugin, you can require both this package and the [Jetpack Autoloader](https://packagist.org/packages/automattic/jetpack-autoloader) in your project's `composer.json` file.
 
-## Usage
+## Overview
 
 The Block Delimiter package provides an efficient, streaming parser for working with WordPress block structure without the memory overhead of `parse_blocks()`. It's designed for scenarios where you need to inspect, find, or modify specific blocks without parsing the entire block tree.
 
+The package includes two classes:
+
+- **`Block_Scanner`** (recommended): A high-performance, mutable scanner that provides the best CPU performance while maintaining near-zero memory overhead.
+- **`Block_Delimiter`** (legacy): A generator-based approach that returns new instances for each delimiter, suitable for existing code that relies on the `scan_delimiters()` interface.
+
+## Block_Scanner (Recommended)
+
+The `Block_Scanner` class is modeled after the WordPress HTML API and provides the best performance characteristics. It mutates itself during scanning and requires a new instance for each document scan.
+
+### Basic Block Scanning
+
+Find and iterate through all block delimiters in a document:
+
+```php
+use Automattic\Block_Scanner;
+
+$post_content = '<!-- wp:paragraph -->
+<p>Hello world!</p>
+<!-- /wp:paragraph -->
+<!-- wp:image {"id":123} -->
+<figure><img src="example.jpg" /></figure>
+<!-- /wp:image -->';
+
+$scanner = Block_Scanner::create( $post_content );
+
+while ( $scanner->next_delimiter() ) {
+    echo "Found block: " . $scanner->get_block_type() . "\n";
+    echo "Type: " . $scanner->get_delimiter_type() . "\n";
+}
+```
+
+**Output:**
+```
+Found block: core/paragraph
+Type: opener
+Found block: core/paragraph
+Type: closer
+Found block: core/image
+Type: opener
+Found block: core/image
+Type: closer
+```
+
+### Finding Specific Block Types
+
+Efficiently find blocks of a specific type without parsing everything:
+
+```php
+use Automattic\Block_Scanner;
+
+$post_content = '<!-- wp:paragraph -->
+<p>Welcome to my blog!</p>
+<!-- /wp:paragraph -->
+<!-- wp:image {"id":456} -->
+<figure><img src="photo.jpg" /></figure>
+<!-- /wp:image -->
+<!-- wp:gallery {"ids":[789,101]} -->
+<figure class="wp-block-gallery">...</figure>
+<!-- /wp:gallery -->';
+
+$scanner = Block_Scanner::create( $post_content );
+
+// Find the first image block
+while ( $scanner->next_delimiter() ) {
+    if ( ! $scanner->is_block_type( 'image' ) ) {
+        continue;
+    }
+    
+    if ( $scanner->opens_block() ) {
+        $attributes = $scanner->allocate_and_return_parsed_attributes();
+        if ( isset( $attributes['id'] ) ) {
+            echo "Found image with ID: " . $attributes['id'];
+            break;
+        }
+    }
+}
```

Comment:
for encouraging more standardized use, we could update this to pass the block type into `opens_block()`

```php
while ( $scanner->next_delimiter() ) {
	if ( ! $scanner->opens_block( 'image' ) ) {
		continue;
	}

	…
}
```

Reply:
I updated the docs in #44365